implants/lib/eldritch/src/file/list_impl.rs

```diff
@@ -0,0 +1,182 @@
+use anyhow::Result;
+use starlark::{values::{dict::Dict, Heap, Value}, collections::SmallMap, const_frozen_string};
+use sysinfo::{System, SystemExt, UserExt};
+use std::fs::DirEntry;
+#[cfg(unix)]
+use std::os::unix::fs::PermissionsExt;
+#[cfg(target_os = "macos")]
+use std::os::macos::fs::MetadataExt;
+#[cfg(target_os = "linux")]
+use std::os::linux::fs::MetadataExt;
+#[cfg(target_os = "windows")]
+use std::os::windows::fs::MetadataExt;
+use chrono::{Utc, DateTime, NaiveDateTime};
+use super::{File, FileType};
+
+const UNKNOWN: &str = "UNKNOWN";
+
+// https://stackoverflow.com/questions/6161776/convert-windows-filetime-to-second-in-unix-linux
+fn windows_tick_to_unix_tick(windows_tick: u64) -> i64{
+    const WINDOWS_TICK: u64 = 10000000;
+    const SEC_TO_UNIX_EPOCH: u64 = 11644473600;
+    return (windows_tick / WINDOWS_TICK - SEC_TO_UNIX_EPOCH) as i64;
+}
+
+fn create_file_from_dir_entry(dir_entry: DirEntry) -> Result<File> {
+    let mut sys = System::new();
+    sys.refresh_users_list();
+
+    let file_name = match dir_entry.file_name().into_string() {
+        Ok(local_file_name) => local_file_name,
+        Err(_) => return Err(anyhow::anyhow!("file.list: Unable to convert file name to string.")),
+    };
+
+    let mut file_type = FileType::Unknown;
+    let tmp_file_type = dir_entry.file_type().expect("file.list: Unable to determine type");
+    if tmp_file_type.is_dir() {
+        file_type = FileType::Directory;
+    } else if tmp_file_type.is_file() {
+        file_type = FileType::File;
+    } else if tmp_file_type.is_symlink() {
+        file_type = FileType::Link;
+    }
+
+    let dir_entry_metadata = dir_entry.metadata().expect("file.list: Unable to get dir_entry metadata.");
```

Comment:
susssssss, im not sure we want to panic here if theres an error.

Reply:
Oh very fair.
I thought this would bubble an error great catch ty nick.